kapitan/refs/env.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2019 The Kapitan Authors
+# SPDX-FileCopyrightText: 2020 The Kapitan Authors <kapitan@google.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"environment refs module"
+
+import base64
+import hashlib
+import os
+
+from kapitan.refs.base import PlainRef, PlainRefBackend
+
+DEFAULT_ENV_REF_VAR_PREFIX = "KAPITAN_VAR_"
+
+
+class EnvRef(PlainRef):
+    def __init__(self, data, **kwargs):
+        """
+        writes plain data, which is the "default" value if the ref cannot be located in the KAPITAN_VAR_*
+        environment vars prefix during the reveal phase.
+        """
+        super().__init__(data, kwargs=kwargs)
+        self.type_name = "env"
+
+    def reveal(self):
+        """
+        Attempt to locate the variable in the environment w/ the suffix.
+        """
+        return os.getenv(
+            "{}{}".format(DEFAULT_ENV_REF_VAR_PREFIX, self.path.split("/")[-1]), default=self.data,
```

Comment:
FWIW this is a very opinionated way to derive the prefix name. I'm open to other suggestions if this isn't very clear. I had also considered converting the path separarator characters to _, but in our our case we use  target_name prefix for a lot of these. I thought that might cause the variable names to get quite lengthly and cause other problems, so I settled on the "last octet" of the path.

Reply:
I'm personally ok with it, it resembles the TF_VAR convention that terraform uses so it makes sense.